internal/proxy/proxy.go

```diff
@@ -259,11 +259,20 @@ func (p *Proxy) https(rw http.ResponseWriter, req *http.Request) {
 func (p *Proxy) connectAndCopy(conn net.Conn, brw *bufio.ReadWriter, req *http.Request, reconnected bool) {
 	logger := req.Context().Value(LogEntryCtx).(*zap.Logger)
 
+	retries := 0
+proxy_dial_retry:
 	// Open connection with downstream proxy
 	pconn, err := net.DialTimeout("tcp", p.config.DownstreamProxyURL.Host, p.config.Timeouts.DownstreamProxy.DialTimeout)
 	if err != nil {
+		if retries < p.config.DownstreamProxyDialRetries {
+			logger.Error("connection to downstream proxy failed.", zap.Error(err))
```

Comment:
If i remembered it correctly, log messages use Upper case. Can you edit it if true? For the log consistency.

Reply:
you mean first letter of the message should be uppercase?

internal/proxy/util.go:                                 c.logger.Error("cannot close write", zap.Error(err))
internal/proxy/proxy.go:                                logger.Error("Panic recovered", zap.Error(err))